2.0/contrib/etc/puma.cfg

```diff
@@ -1,4 +1,40 @@
+def get_max_memory()
+    return ENV['MEMORY_LIMIT_IN_BYTES'].to_i if ENV.has_key? 'MEMORY_LIMIT_IN_BYTES'
+
+    # Assume unlimited memory. 0.size is the number of bytes a ruby
+    # Fixnum class can hold. One bit is used for sign and one is used
+    # by Ruby to determine whether it's a number or pointer to an object.
+    # That's why we subtract two bits. This expresion should therefore be
+    # the largest signed Fixnum possible.
+    (2 ** (8*0.size - 2) - 1)
+end
+
+def get_min_threads()
+    ENV.fetch('PUMA_MIN_THREADS', '0').to_i
+end
+
+def get_max_threads()
+    ENV.fetch('PUMA_MAX_THREADS', '16').to_i
+end
+
+# Determine the maximum number of workers that are allowed by the available
+# memory. Puma documentation recommends the maximum number of workers to be
+# set to the number cores.
+def get_workers()
+    return ENV['PUMA_WORKERS'].to_i if ENV.has_key? 'PUMA_WORKERS'
+
+    base_memory = 50 * 1024 * 1024
+    per_worker_base_memory = 15 * 1024 * 1024
+    per_thread_memory = 103 * 1024 # roughly 0.1 MiB
```

Comment:
Very well. 128 it is. (I took a quick glance at the data and 128 should be all right)